### DISCUSSION TITLE
different strcturalSharing option for raw response data and selector data

### DISCUSSION THREAD

**Discussion by bartzy** [TanStack/query#2912] [Ideas]
Hi!

Does it make sense to add an option to not deep-equal check the selector data if the raw response data is shared?
My main goal is to avoid running the deep-equal checks twice when I know the selectors will never change the data if the input (response data) didn't change.

**Comment by TkDodo** [comment]
> My main goal is to avoid running the deep-equal checks twice when I know the selectors will never change the data if the input (response data) didn't change.

That won't happen if you memoize the selector.

- First, the data returned from the network will be structurally shared / deep equal replaced, then it will be put into the cache
- `select` then only runs if that data is different, or if the select function itself changed
- only then, the result of `select` is again structurally shard and deep equal replaced.

If you have an inline function for select, memoize it with useCallback. If it doesn't have dependencies, you can also consider moving it out of the render function.

See: https://tkdodo.eu/blog/react-query-data-transformations#3-using-the-select-option
